### PR TITLE
Add gh-pages

### PR DESCRIPTION
**Pull request by FredrikAugust**
Here's the website so far, contribute at fullstck/open-source-ideas-website.

### CODE REVIEW

**Comment by Haroenv** [comment]
The README is conflicting. You should merge that from mikaelbr/master or leave the readme out on the gh-pages branch

**Comment by FredrikAugust** [comment]
Yeah, see that now. Will remove it, one sec.

**Comment by FredrikAugust** [comment]
There we go

**Comment by Haroenv** [comment]
Because you're merging from `gh-pages` on your own repo, interested people can see it on https://fredrikaugust.github.io/open-source-ideas/

**Comment by FredrikAugust** [comment]
Yep, just create an issue if you're missing anything.

**Comment by FredrikAugust** [comment]
Interestingly enough, GitHub treats PRs as issues, so this PR is showing up on the website. Don't know if that's something we want or not. Won't be a big problem anyway since there won't be many PRs to the repo.

**Comment by mikaelbr** [comment]
Could you possibly try to use [`_docs` folder](https://github.com/blog/2228-simpler-github-pages-publishing) instead of `gh-pages` branch? I think it's easier to maintain.

**Comment by Haroenv** [comment]
I'm not sure why the `remove_markdown` is needed. IMO it's better if both the home page and the have parsed markdown.

**Comment by FredrikAugust** [comment]
@Haroenv Because the cards would take up way more space then, and it's not really necessary IMO.

**Comment by FredrikAugust** [comment]
@mikaelbr How is that more maintainable? Isn't it better to keep website and the rest separate?

**Comment by Haroenv** [comment]
https://fredrikaugust.github.io/open-source-ideas/issue.html?issue=8 for example (probably because it's parsed as plain markdown, and not GFM) looks kinda weird to me, both in the detail view and in the overview. Maybe the text should be truncated but still parsed?

**Comment by FredrikAugust** [comment]
Yeah, that's true. If we're going to reduce the output we might as well only show the titles on the main page. As far as GFM goes I'll look into it later.

**Comment by FredrikAugust** [comment]
https://github.com/evilstreak/markdown-js <- I'm using this now, and it doesn't support GFM :/

**Comment by Haroenv** [comment]
What could be an option is requiring a certain format that is a short description first, then `---` after which more detail is written. On the website only the part before `---` would be shown.

**Comment by FredrikAugust** [comment]
Yeah, that'd work. Looking at [this](http://strapdownjs.com/) now. Seems quite nice, also supports GFM

**Comment by Haroenv** [comment]
https://github.com/chjj/marked seems to get quite a lot of good attention, and also supports GFM

**Comment by FredrikAugust** [comment]
Yeah, that seems better. Will add it soon(tm). Just got to address an issue with my own website first :)

**Comment by FredrikAugust** [comment]
There we go, now it's using marked on index page and issue page.

**Comment by Haroenv** [comment]
Looks great! I've got some ideas for improvement, but this seems like a good start to *me*

**Comment by FredrikAugust** [comment]
Yep, not meant as a finished product, just thought it would be nice to get the base implementation down :)

**Comment by FredrikAugust** [comment]
Is there anything that should be done before merging this?

**Comment by mikaelbr** [comment]
@FredrikAugust My thinking on `_docs` vs `gh-pages`: I'm not firm on this, in any way, but I usually prefer to have things like this as close to "the source" as possible, and not spread across different places. And there would be no managing long living parallel branches (becomes an "issue" when there are some common files a cross branches).

**Comment by FredrikAugust** [comment]
What files would be living across branches here? They are basically two different projects, hence putting them in two places

**Comment by Haroenv** [comment]
A logo image, or the image now on the README

**Comment by FredrikAugust** [comment]
Yeah, good point. Will fix tomorrow :)

**Comment by mikaelbr** [comment]
Woops. Had forgotten to publish my review. Did this now

File: js/main.js [hunk 1]
```diff
@@ -0,0 +1,59 @@
+// This is where the magic happens
+// Written by Fredrik August Madsen-Malmo (github@fredrikaugust)
+
+var URL = "https://api.github.com/repos/mikaelbr/open-source-ideas/issues";
+
+function display_issues (flags = []) {
```
Comment (FredrikAugust) [inline js/main.js]: Does JS have `||=`?

File: tags [hunk 2]
```diff
@@ -0,0 +1,10 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
+!_TAG_FILE_SORTED	1	/0=unsorted, 1=sorted, 2=foldcase/
+!_TAG_PROGRAM_AUTHOR	Darren Hiebert	/dhiebert@users.sourceforge.net/
+!_TAG_PROGRAM_NAME	Exuberant Ctags	//
+!_TAG_PROGRAM_URL	http://ctags.sourceforge.net	/official site/
+!_TAG_PROGRAM_VERSION	5.8	//
+#cards 	css/style.css	/^#cards {$/;"	i
+URL	js/main.js	/^var URL = "https:\/\/api.github.com\/repos\/mikaelbr\/open-source-ideas\/issues";$/;"	v
+display_issues	js/main.js	/^function display_issues (flags = []) {$/;"	f
+show_issues_in_dom	js/main.js	/^function show_issues_in_dom (issues) {$/;"	f
```
Comment (FredrikAugust) [inline tags]: Oops

File: tags [hunk 3]
```diff
@@ -0,0 +1,10 @@
+!_TAG_FILE_FORMAT	2	/extended format; --format=1 will not append ;" to lines/
```
Comment (FredrikAugust) [inline tags]: Ctags :) Really nice for navigating code, using it with a plugin in vim

File: js/main.js (hunk 1 above)
Comment (mikaelbr) [inline js/main.js]: Technically, it should bore correctly be written as

```js
flags = typeof flags === 'undefined' ? [] : flags;
```

But `flags = flags || [];` is a often used "hack"/shortcut to in many cases achieve the same effect as in javascript boolean expression return the first truthy value.

---

```js
var default1 = function (flags) {
  return typeof flags === 'undefined' ? [] : flags;
};
var default2 = function (flags) {
  return flags || [];
};

```

Here `default1` and `default2` isn't strictly the same when it comes to falsey values. See details:

```js
default1(); //> []
default1(undefined); //> []
default1(null); //> null
default1(false); //> false
default1(''); //> ''

// But with default2:
default1(); //> []
default1(undefined); //> []
default1(null); //> []
default1(false); //>  []
default1(''); //> []
```

File: tags (hunk 3 above)
Comment (mikaelbr) [inline tags]: I see! Got it. So I figured it should have been in `.gitignore` then. (I see you already realised it in https://github.com/mikaelbr/open-source-ideas/pull/24#discussion_r85046228)

**Comment by FredrikAugust** [comment]
Yep, started to implement sorting by tags (flags), but didn't finish. Don't
think "a lot" of browsers will crash, but perhaps some.

The file is a ctags file, using it to navigate my code in vim :) Really
handy, would recommend looking into it.

**Comment by FredrikAugust** [comment]
Didn't, will remove

**Comment by FredrikAugust** [comment]
Think I'll be able to fix the issues today, but I've lost my charger so can't do it before earliest 16:00 (norwegian time)

**Comment by mikaelbr** [comment]
No worries!

**Pull request closed without merging**